manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -37,32 +40,34 @@ type DataRequirements struct {
 	// +required
 	Interface InterfaceDetails `json:"interface"`
 
-	// CopyRequrements include the requirements for copying the data
+	// CopyRequrements include the requirements for explicit requests to copy the data
 	// +optional
-	Copy CopyRequirements `json:"copy,omitempty"`
+	FlowParams FlowRequirements `json:"flowParams,omitempty"`
 }
 
-// DataContext indicates data set chosen by the Data Scientist to be used by his application,
-// and includes information about the data format and technologies used by the application
-// to access the data.
+// DataContext indicates data set being processed by the workload
+// and includes information about the data format and technologies used to access the data.
 type DataContext struct {
-	// DataSetID is a unique identifier of the dataset chosen from the data catalog for processing by the data user application.
+	// DataSetID is a unique identifier of the dataset chosen from the data catalog.  For data catalogs that support multiple sub-catalogs, it includes the catalog id and the dataset id.
+	// When writing a new dataset it is the name provided by the user or workload generating it, and should include the sub-catalog name where appropriate.
 	// +required
 	// +kubebuilder:validation:MinLength=1
 	DataSetID string `json:"dataSetID"`
 
-	// CatalogService represents the catalog service for accessing the requested dataset.
-	// If not specified, the enterprise catalog service will be used.
+	// Flows indicates what is being done with the particular dataset - ex: read, write, copy, delete
+	// If more than one flow is indicated, the order is respected.
+	// This is optional for the purpose of backward compatability.  If nothing is provided, read is assumed.
 	// +optional
-	CatalogService string `json:"catalogService,omitempty"`
+	Flows []DataFlow `json:"flows,omitempty"`
+
 	// Requirements from the system
 	// +required
 	Requirements DataRequirements `json:"requirements"`
 }
 
 // FybrikApplicationSpec defines data flows needed by the application, the purpose and other contextual information about the application.
-// Read flow - if selector is populated, fybrik builds a data plane for reading the specified data sets
-// Ingest flow - if no selector, and data/copy/required is true then the data specified is copied into a bucket allocated by fybrik and is cataloged in the data catalog
+// Read flow - if selector is populated, fybrik builds a data plane for reading the specified data sets.
+// Ingest flow - if no selector, and data/copy/required is true then the data specified is copied into a bucket allocated by fybrik and is cataloged in the data catalog.
 type FybrikApplicationSpec struct {
```

Comment:
why do we still need this selector based separation?

Reply:
Old comments removed